src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/WindowBackdrop.cs

```diff
@@ -0,0 +1,267 @@
+
+            case WindowBackdropType.TransientWindow:
+                return ApplyDwmWindowAttribute(hWnd, Standard.DWMSBT.DWMSBT_TRANSIENTWINDOW);
+
+            case WindowBackdropType.TabbedWindow:
+                return ApplyDwmWindowAttribute(hWnd, Standard.DWMSBT.DWMSBT_TABBEDWINDOW);
+        }
+
+        return ApplyDwmWindowAttribute(hWnd, Standard.DWMSBT.DWMSBT_NONE);
+    }
+
+    /// <summary>
+    /// Tries to remove backdrop effects if they have been applied to the <see cref="Window"/>.
+    /// </summary>
+    /// <param name="window">The window from which the effect should be removed.</param>
+    internal static bool RemoveBackdrop(System.Windows.Window window)
+    {
+        if (window is null)
+        {
+            return false;
+        }
+
+        IntPtr windowHandle = new WindowInteropHelper(window).Handle;
+
+        return RemoveBackdrop(windowHandle);
+    }
+
+    /// <summary>
+    /// Tries to remove all effects if they have been applied to the <c>hWnd</c>.
+    /// </summary>
+    /// <param name="hWnd">Pointer to the window handle.</param>
+    internal static bool RemoveBackdrop(IntPtr hWnd)
+    {
+        if (hWnd == IntPtr.Zero)
+        {
+            return false;
+        }
+
+        _ = RestoreContentBackground(hWnd);
+
+        var backdropPvAttribute = Standard.DWMSBT.DWMSBT_NONE;
+        var dwmResult = NativeMethods.DwmSetWindowAttributeSystemBackdropType(hWnd, backdropPvAttribute);
+        return dwmResult == HRESULT.S_OK;
+    }
+
+    /// <summary>
+    /// Tries to remove background from <see cref="Window"/> and it's composition area.
+    /// </summary>
+    /// <param name="window">Window to manipulate.</param>
+    /// <returns><see langword="true"/> if operation was successful.</returns>
+    internal static bool RemoveBackground(System.Windows.Window window)
+    {
+        if (window is null)
+        {
+            return false;
+        }
+
+        // Remove background from visual root
+        window.SetCurrentValue(System.Windows.Controls.Control.BackgroundProperty, Brushes.Transparent);
+
+        IntPtr windowHandle = new WindowInteropHelper(window).Handle;
+
+        if (windowHandle == IntPtr.Zero)
+        {
+            return false;
+        }
+
+        var windowSource = HwndSource.FromHwnd(windowHandle);
+
+        // Remove background from client area
+        if (windowSource?.CompositionTarget != null)
+        {
+            windowSource.CompositionTarget.BackgroundColor = Colors.Transparent;
+        }
+
+        return true;
+    }
+
+    private static bool UpdateGlassFrame(IntPtr hWnd, WindowBackdropType backdropType)
+    {
+        if (hWnd == IntPtr.Zero)
+        {
+            return false;
+        }
+
+        MARGINS margins = new MARGINS();
+        if(backdropType != WindowBackdropType.None)
+        {
+            margins = new MARGINS { cxLeftWidth = -1, cxRightWidth = -1, cyTopHeight = -1, cyBottomHeight = -1 };                    
```

Comment:
Are we certain that we want to hard code these numbers? Using `-1` will slightly draw the title bar buttons into the client area in full screen, which might be unexpected and unwanted:
![image](https://github.com/dotnet/wpf/assets/1675889/d91a96af-6d8a-4ce2-ad29-76b16d006651)

An alternative is to use `margins = new MARGINS { cyBottomHeight = int.MaxValue }`. That way the title bar buttons stay the same height as the title bar which matches the current WPF behavior in full screen:
![image](https://github.com/dotnet/wpf/assets/1675889/38519ada-2e1a-4d5c-831a-5c21a5e3bf83)

Just as a thought to maybe add an opt-out option to this behavior.

Reply:
On which OS version did you test this?
When doing so on Win11 23H2 the caption buttons disappear.
At least with a custom window chrome.

Reply:
Yes, observed this on win11 23H2 on multiple machines. Looks like by design.

I cannot speak for custom window chrome. Is this even drawn by the OS?

Reply:
Custom window chrome usually just means claiming the titlebar area for drawing, for example WPF content.
The WindowChrome class of WPF allows this, for example.

Reply:
Indeed `WindowChrome` behaves differently from the way the OS would draw the title bar when you use this trick. But even without `DwmExtendFrameIntoClientArea`, using `WindowChrome` results in a different title bar than the OS would draw. 
In general, `WindowChrome` is considered broken for quite some time now, so I'm not really surprised though.

Reply:
It does not result in a different titlebar than the OS would draw, depending on the flags you use on WindowChrome (or any other class/library that supports it).

WindowChrome is still a supported part of WPF.
If the team want's to abandon that class they should announce that and remove it.
If it is still supported it should work when using the backdrop and that means the backdrop should not even set those margins, but it should be removed here and from WindowChrome and be moved to Window.

Reply:
I don't see any option to remove the right margin next to the close button:
![image](https://github.com/dotnet/wpf/assets/1675889/57e43622-2070-4a97-82bb-1b6722d11371)
No other application window behaves that way, therefore it is not the same.

And I never said it's not supported or should be abandoned, just said it's broken.

As of now it's pure luck if the glass effect works when you use it together with `WindowChrome` as both use `DwmExtendFrameIntoClientArea` in a different way and it is just luck which one calls it last.

This discussion loses the focus, my argument is that the title bar buttons draw into the client area in full screen which was not the case without the Mica effect and I wanted to discuss whether we are OK with that or should think about enabling also the previous behavior.

Reply:
@MichaeIDietrich, I decided to set the Thickness = -1 after seeing the GlassFrameThickness coercion method, where if there is any negative value in thickness, we coerce it to -1, which allows us to enable the glass effect.
 
I see that how setting bottomMargin to max value helps in enabling the glass effect without the unintended effect of client area controls overlapping the caption buttons. But I will have to look more into it to fix this behavior.
 
> As of now it's pure luck if the glass effect works when you use it together with WindowChrome as both use DwmExtendFrameIntoClientArea in a different way and it is just luck which one calls it last.
 
I had this at the back of my mind, that it will cause issues with WindowChrome calling DwmExtendIntoClientArea.
 
Regarding the opt-out although it is not present now, but we plan on adding it. This call was added here for enabling the glass effect when applying backdrop. 
And we want to make backdrop as opt-in so it will be handled by setting the property on window.
 
 
@batzen, as I mentioned above, the decision of choosing -1 came from the WindowChrome glass frame coercion.
 
I know that WindowChrome has it's share of bugs and that needs to be fixed. WindowChrome is not abandoned. Fixing WindowChrome is on our bucket list as that will allow developers to create customizable title bar and that's part of Win11 signature experiences.

Reply:
@MichaeIDietrich 

> I don't see any option to remove the right margin next to the close button:

The window with space in right margin is the one when WindowChrome is enabled along with Win11 theme enabled ?

Reply:
> @MichaeIDietrich, I decided to set the Thickness = -1 after seeing the GlassFrameThickness coercion method, where if there is any negative value in thickness, we coerce it to -1, which allows us to enable the glass effect.
> 
> I see that how setting bottomMargin to max value helps in enabling the glass effect without the unintended effect of client area controls overlapping the caption buttons. But I will have to look more into it to fix this behavior.

It's also worth mentioning that the issue itself is known and not WPF but Win32 specific:
https://github.com/microsoft/WindowsAppSDK/issues/180#issuecomment-698500844

I think using `-1` is the common way of enabling the glass effect on the window background, which most probably everyone uses when enabling this effect. 
I just stumbled upon this when we enabled this effect in our business applications that we suddenly had this unintended overlapping there. I think the workaround I showed is not really known nor documented, so I'm not sure whether it is save to rely on it but I can tell it works without issues so far. 
(_We do not use `WindowChrome`, so we just ignore that it is not compatible to this workaround_).

So maybe someone from the Windows dev team could help out, what's best to do here.

> The window with space in right margin is the one when WindowChrome is enabled along with Win11 theme enabled ?

The margin on the right side is not new with Windows 11 theming. The screenshot is from `.NET 8.0.3 WPF` on Windows 11 only using `WindowChrome`. If I remember correctly the same issue already existed on Windows 10, so it is not directly connected to your changes but to the overall missing updates to `WindowChrome`.

Reply:
Why add a context switch?
The whole theme is opt-in as of now, isn't it?

I know that -1 is the required value to get the backdrop behind the whole window, but it shouldn't be the responsibility of the backdrop code to set the glass frame thickness as that will interfere with code that sets different values. What should happen if i want to disable backdrop during runtime? Will you remove the glass frame?
That's why i suggested separating backdrop and glass frame thickness to separate window properties. That's also exactly what ControlzEx does.

Regarding glass frame thickness:
I already said that using int.max is not an acceptable workaround as it moved the window buttons completely off screen or even crashed the application.
You can't use it properly without also handling nccalcsize, as far as i found out.
Using the window chrome from controlzex with glass frame of -1 and a backdrop does not cause that margin issue.

Reply:
> Why add a context switch?

I thought of doing so because of the interference as you mentioned above. Some code will be using WindowChrome or added a property to do so and there can be interference there, producing weird results. So, context switch would enable the scenario where developers can disable the backdrop and still use the theme. This is a temprorary switch and I will remove it when we make the WindowBackdrop ( or some similar named ) property public.

> but it shouldn't be the responsibility of the backdrop code to set the glass frame thickness as that will interfere with code that sets different values.

> What should happen if i want to disable backdrop during runtime? Will you remove the glass frame?

In the current approach where I am moving WindowBackdrop to OnSourceInitialized and using a seperate property ( internal for now ) I am resetting the glass frame thickness. 

> That's why i suggested separating backdrop and glass frame thickness to separate window properties. That's also exactly what ControlzEx does.

Regarding this, we will have to reason out a bit. On one hand it makes thing clearer, but on the other hand it is bringing responsibility from already existing WindowChrome to Window. 

> You can't use it properly without also handling nccalcsize, as far as i found out.

Yeah, I found this too. We will need to handle the NCCALCSIZE message to fix this issue.